diagnostic plot for normal likelihood and normal prior mixture

LSun opened the pull request:
diag.ash(ash_obj) produces a Q-Q plot for estimated Fhat vs uniform, which should look like a straight line from 0 to 1 under goodness of fit.
please check it makes sense.
can add more cosmetic effect to the plot later.

stephens999 commented on R/ashutility.R:
```diff
@@ -57,6 +57,28 @@ plot.ash = function(x,...,xmin,xmax){
+                 main = "Diagnostic Plot for ASH",
+                 ...)
+  abline(0, 1, lty = 2, col = "red")
+}
```
can you return(cdfhat) in case someone wants to get access to it?

stephens999 commented on R/ashutility.R:
```diff
@@ -57,6 +57,28 @@ plot.ash = function(x,...,xmin,xmax){
+#'
+#' @export
+#'
+diag.ash = function(x, ...){
```
can we call it plot_diagnostic ?

LSun commented:
removed copied code from plot_diagnostic, used my own simpler implementation
implemented uniform prior + normal likelihood
uniform prior + t likelihood not yet implemented, can be implemented in future

stephens999 commented:
just two things:
i) can you add plot.it=TRUE flag to the plot function? Then if it is false don't actually plot the plot? That would allow someone to use this function to compute the cdf values only if they want to.
ii) any particularly reason you return it as a list() with a single element, rather than just return a matrix? If not, can you just return a matrix?

LSun commented:
i) sure it's done.
ii) i just want to give the output a name.  no particular reasons, just thought a name would be informative i guess?  now it's removed and the function just returns a matrix.
